File: app/frontend/src/lib/screens.ts
// Static metadata for the 4 gated screens (ACRI-66). Each entry is used by
// `RequireAuth`-wrapped routes and by `StubScreen` to know which backend
// stub endpoint to call. No business logic lives here — real per-screen
// behavior lands in later, dedicated stories.

export interface ScreenMeta {
  id: string;
  path: string;
  label: string;
  backendPath: string;
}

export const SCREENS: readonly ScreenMeta[] = [
  {
    id: "risk-dashboard",
    path: "/risk-dashboard",
    label: "Risk Dashboard",
    backendPath: "/screens/risk-dashboard",
  },
  {
    id: "ingredient-detail",
    path: "/ingredient-detail",
    label: "Ingredient Detail",
    backendPath: "/screens/ingredient-detail",
  },
  {
    id: "chat-agent",
    path: "/chat-agent",
    label: "Chat Agent",
    backendPath: "/screens/chat-agent",
  },
  {
    id: "purchase-order-draft",
    path: "/purchase-order-draft",
    label: "Purchase Order Draft",
    backendPath: "/screens/purchase-order-draft",
  },
];

File: app/backend/routes/screens.py
"""Protected stub screen routes (ACRI-66).

Four placeholder routes, one per gated screen named in the approved Design
Document (risk dashboard, ingredient detail, Chat Agent, purchase-order
draft). Each requires authentication only (`get_current_user`) — there is no
persona-based branching anywhere in this module, so both personas receive an
identical response shape (AC5). Real business logic for each screen lands in
later, dedicated stories.
"""

from __future__ import annotations

from fastapi import APIRouter, Depends

from db.models import User
from middleware.auth import get_current_user
from schemas.auth import ScreenPlaceholderResponse, UserOut

router = APIRouter(prefix="/screens", tags=["screens"])

_PLACEHOLDER_MESSAGE = "Authenticated placeholder — business logic not yet implemented"


def _placeholder_response(screen: str, current_user: User) -> ScreenPlaceholderResponse:
    return ScreenPlaceholderResponse(
        screen=screen,
        message=_PLACEHOLDER_MESSAGE,
        user=UserOut(email=current_user.email, persona=current_user.persona),
    )


@router.get("/risk-dashboard", response_model=ScreenPlaceholderResponse)
def risk_dashboard(current_user: User = Depends(get_current_user)) -> ScreenPlaceholderResponse:
    return _placeholder_response("risk-dashboard", current_user)


@router.get("/ingredient-detail", response_model=ScreenPlaceholderResponse)
def ingredient_detail(current_user: User = Depends(get_current_user)) -> ScreenPlaceholderResponse:
    return _placeholder_response("ingredient-detail", current_user)


@router.get("/chat-agent", response_model=ScreenPlaceholderResponse)
def chat_agent(current_user: User = Depends(get_current_user)) -> ScreenPlaceholderResponse:
    return _placeholder_response("chat-agent", current_user)


@router.get("/purchase-order-draft", response_model=ScreenPlaceholderResponse)
def purchase_order_draft(
    current_user: User = Depends(get_current_user),
) -> ScreenPlaceholderResponse:
    return _placeholder_response("purchase-order-draft", current_user)
